### PR TITLE
add wrapper to support multiple p2p plugins

### PR DESCRIPTION
**Pull request by Sergei-Lebedev**

### CODE REVIEW

**Comment by mellanox-github** [comment]
<details><summary>Mellanox CI status: <b>PASS</b#  > (click for details)</summary>

**Note:** the artefacts will be deleted after 20-Mar-2020

Artefact file | Status
-- | --
[build_nccl_rdma_sharp_plugins.log](https://gist.githubusercontent.com/mellanox-github/42c0b5f411365ff4a8dc70569026660e/raw/549f9d79fbc61d1fd9eec2cbaa7b6f7b113dde6c/build_nccl_rdma_sharp_plugins.log) | :heavy_check_mark: PASS
[configure_sharp_startup.log](https://gist.githubusercontent.com/mellanox-github/42c0b5f411365ff4a8dc70569026660e/raw/017bc5006373d0ae97e62a35b6dc9a781478bbb1/configure_sharp_startup.log) | :heavy_check_mark: PASS
[configure_sharp_stop.log](https://gist.githubusercontent.com/mellanox-github/42c0b5f411365ff4a8dc70569026660e/raw/db286e03a0d2398131301a0fc7c9d03eef27c6a2/configure_sharp_stop.log) | :heavy_check_mark: PASS
[run_nccl_test.log](https://gist.githubusercontent.com/mellanox-github/42c0b5f411365ff4a8dc70569026660e/raw/2db43608dc10bd4d3f61111111b8886336310caa/run_nccl_test.log) | :heavy_check_mark: PASS

**Comment by mellanox-github** [comment]
Mellanox CI PASSed.

File: configure.ac [hunk 1]
```diff
@@ -77,6 +77,27 @@ AC_CHECK_LIB([ibverbs], [ibv_get_device_list],
 AC_CHECK_DECLS(IBV_ACCESS_RELAXED_ORDERING, [], [],
                     [[#include <infiniband/verbs.h>]])
 
+#check for UCX
+AC_ARG_WITH([ucx],
+        AC_HELP_STRING([--with-ucx=PATH], [Path to UCX installation]),
+        [AS_IF([test "x$with_ucx" != xno],
+          [AC_SUBST([UCX_PATH], $with_ucx)
```
Comment (bureddy) [inline configure.ac]: may be users might just compile w/o any path as UCX is shipped with Linux distros. ucx libs are in /usr/lib64/libucp.so.0

Comment (bureddy) [reply configure.ac]: need to support that as well

File: src/p2p_plugin.c [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+#include "nccl_net.h"
+
+extern ncclNet_t ucxPlugin;
+extern ncclNet_t ibPlugin;
+ncclDebugLogger_t pluginLogFunction;
+
+ncclResult_t pluginInit(ncclDebugLogger_t logFunction);
+
+ncclNet_t NCCL_PLUGIN_SYMBOL = {
+  "NCCL RDMA SHARP Plugin",
```
Comment (bureddy) [inline src/p2p_plugin.c]: remove SHARP here

File: src/p2p_plugin.c [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+  NULL
+};
+
+
+ncclResult_t pluginInit(ncclDebugLogger_t logFunction){
+  pluginLogFunction = logFunction;
+  NCCL_PLUGIN_SYMBOL = ibPlugin;
+#ifdef HAVE_UCX_PLUGIN
+  const char *p2pLayer = getenv("NCCL_PLUGIN_P2P");
+  if ((p2pLayer != NULL) && !strcasecmp(p2pLayer, "ucx"))  NCCL_PLUGIN_SYMBOL = ucxPlugin;
```
Comment (bureddy) [inline src/p2p_plugin.c]: check if option value other than "ib" and "ucx" and report error.

**Comment by mellanox-github** [comment]
Mellanox CI FAILed.

**Comment by artemry-mlnx** [comment]
The CI cluster is under maintenance (SW upgrade). I'll inform when the cluster will be ready.

**Comment by artemry-mlnx** [comment]
bot:mlx:test

**Comment by mellanox-github** [comment]
Mellanox CI FAILed.

**Comment by artemry-mlnx** [comment]
Need to update the CI scripts after CI cluster upgrade - I'll do this.

**Comment by mellanox-github** [comment]
<details><summary>Mellanox CI status: <b>PASS</b#  > (click for details)</summary>

**Note:** the artefacts will be deleted after 31-Mar-2020

Artefact file | Status
-- | --
[build_nccl_rdma_sharp_plugins.log](https://gist.githubusercontent.com/mellanox-github/0093ab91492e68206e94e5fd3af5de97/raw/69eb87c9fac8987ee3ae36b9017c13ba361f8245/build_nccl_rdma_sharp_plugins.log) | :heavy_check_mark: PASS
[configure_sharp_startup.log](https://gist.githubusercontent.com/mellanox-github/0093ab91492e68206e94e5fd3af5de97/raw/9fcadfc27d7e8c6ec73327c57ad1cb6f3f778c74/configure_sharp_startup.log) | :heavy_check_mark: PASS
[configure_sharp_stop.log](https://gist.githubusercontent.com/mellanox-github/0093ab91492e68206e94e5fd3af5de97/raw/1629c94f6d15ef09f1b97645fa2211f22ecce34e/configure_sharp_stop.log) | :heavy_check_mark: PASS
[run_nccl_test.log](https://gist.githubusercontent.com/mellanox-github/0093ab91492e68206e94e5fd3af5de97/raw/07f5c53053ae0fda4118a5083b98c4a0893c8cce/run_nccl_test.log) | :heavy_check_mark: PASS

**Comment by mellanox-github** [comment]
Mellanox CI PASSed.

**Pull request merged**